docs/synapse/deploymentguide.rst

```diff
@@ -0,0 +1,355 @@
+
+In the Synapse service architecture, an Axon provides a place to store arbitrary bytes/files as binary
+blobs and exposes APIs for streaming files in and out regardless of their size.  Given sufficient file system
+size, an Axon can be used to efficiently store and retrieve very large files as well as a high number
+(easily billions) of files.
+
+**Inside the AHA container**
+
+Generate a one-time use provisioning URL::
+
+    python -m synapse.tools.aha.provision.service 00.axon
+
+You should see output that looks similar to this::
+
+    one-time use URL: ssl://aha.loop.vertex.link:27272/<guid>?certhash=<sha256>
+
+**On the Host**
+
+Create the container directory::
+
+    mkdir -p /srv/syn/axon/storage
+    chown -R 999 /srv/syn/axon/storage
+
+Create the ``/srv/syn/00.axon/docker-compose.yaml`` file with contents::
+
+    version: "3.3"
+    services:
+      00.axon:
+        user: 999
+        image: vertexproject/synapse-axon:v2.x.x
+        network_mode: host
+        restart: unless-stopped
+        volumes:
+        - ./storage:/vertex/storage
+        environment:
+            - SYN_AXON_AHA_PROVISION=ssl://aha.loop.vertex.link:27272/<guid>?certhash=<sha256>
+
+.. note::
+
+    Don't forget to replace your one-time use provisioning URL!
+
+Start the container::
+
+    docker-compose --file /srv/syn/axon/docker-compose.yaml pull
+    docker-compose --file /srv/syn/axon/docker-compose.yaml up -d
+
+Deploy JSONStor Service
+=======================
+
+**Inside the AHA container**
+
+Generate a one-time use provisioning URL::
+
+    python -m synapse.tools.aha.provision.service 00.jsonstor
+
+You should see output that looks similar to this::
+
+    one-time use URL: ssl://aha.loop.vertex.link:27272/<guid>?certhash=<sha256>
+
+**On the Host**
+
+Create the container directory::
+
+    mkdir -p /srv/syn/00.jsonstor/storage
+    chown -R 999 /srv/syn/00.jsonstor/storage
+
+Create the ``/srv/syn/00.jsonstor/docker-compose.yaml`` file with contents::
+
+    version: "3.3"
+    services:
+      00.jsonstor:
+        user: 999
+        image: vertexproject/synapse-jsonstor:v2.x.x
+        network_mode: host
+        restart: unless-stopped
+        volumes:
+        - ./storage:/vertex/storage
+        environment:
+            - SYN_JSONSTOR_AHA_PROVISION=ssl://aha.loop.vertex.link:27272/<guid>?certhash=<sha256>
+
+.. note::
+
+    Don't forget to replace your one-time use provisioning URL!
+
+Start the container::
+
+    docker-compose --file /srv/syn/00.jsonstor/docker-compose.yaml pull
+    docker-compose --file /srv/syn/00.jsonstor/docker-compose.yaml up -d
+
+Deploy Cortex Service
+=====================
+
+**Inside the AHA container**
+
+Edit or copy the following contents to the file ``/tmp/cortex.yaml`` inside the container::
+
+    axon: aha://axon...
+    jsonstor: aha://jsonstor...
+
+Generate a one-time use provisioning URL::
+
+    python -m synapse.tools.aha.provision.service 00.cortex --user root --cellyaml /tmp/cortex.yaml
```

Comment:
When I did this my user certs are for root but the auto-generated aha:registry url uses cortex as the user.  Once I switched that root it worked (and changed the port).

Reply:
Modifying URL generation to allow aha:user to win during provisioning, thanks!